etl/steps/data/garden/iea/2024-07-04/critical_minerals.py

```diff
@@ -0,0 +1,359 @@
+    # But the "Other uses" will be added to all minerals later.
+    # Hence, remove other uses and total demand.
+    tb_graphite_total = tb_graphite_total[
+        ~tb_graphite_total["technology"].isin([TOTAL_DEMAND_LABEL, OTHER_LABEL])
+    ].reset_index(drop=True)
+
+    # Calculate the total graphite demand in the 4.x sheets.
+    tb_graphite_demand = (
+        tb_demand[tb_demand["mineral"].isin(graphite_uses)].drop(columns=["mineral"]).reset_index(drop=True)
+    )
+    combined = tb_graphite_demand.merge(
+        tb_graphite_total, how="inner", on=["case", "scenario", "technology", "year"], suffixes=("", "_summary")
+    )
+    # Check that the totals in sheet 1 are always larger than the totals calculated from combining the 4.x sheets (or within 1%).
+    error = "Sheet on clean technology graphite demand by mineral should be >= the sum of the graphite demands from the 4.x sheets."
+    assert combined[(combined["demand_summary"] < (combined["demand"] * 0.99))].empty, error
+    combined["difference"] = combined["demand_summary"] - combined["demand"]
+
+    # Create a new technology that contains all the remaining graphite demand not included in the 4.x sheets.
+    # NOTE: Only include it if the difference is larger than 1%.
+    tb_graphite_other = (
+        combined[combined["difference"] >= combined["demand"] * 0.01]
+        .assign(**{"mineral": OTHER_GRAPHITE_LABEL})
+        .drop(columns=["demand", "demand_summary"])
+        .rename(columns={"difference": "demand"})
+        .reset_index(drop=True)
+    )
+
+    # Append the demand from other low emissions power generation to the demand table.
+    tb_demand_with_graphite_other = pr.concat([tb_demand, tb_graphite_other], ignore_index=True)
+
+    return tb_demand_with_graphite_other
+
+
+def prepare_total_demand_table(ds_meadow: Dataset) -> Table:
+    # Read table on total demand of minerals (including uses outside of clean tech).
+    tb_total = ds_meadow["demand_for_key_minerals"].reset_index()
+
+    # Sanity checks.
+    total_clean_label = "Total clean technologies"
+    assert total_clean_label in set(tb_total["technology"]), f"'{total_clean_label}' not found in sheet 1."
+    assert not tb_total[
+        (tb_total["technology"].str.startswith("Share of"))
+    ].empty, "Rows for 'Share of...' not found in sheet 1."
+
+    # Remove rows of shares and total clean technology demand (but keep grand total).
+    tb_total = tb_total[
+        (tb_total["technology"] != total_clean_label) & (~tb_total["technology"].str.startswith("Share of"))
+    ].reset_index(drop=True)
+
+    # Even if not explicitly mentioned, it seems that "Base case" is assumed in this sheet.
+    tb_total["case"] = "Base case"
+
+    return tb_total
+
+
+def add_demand_from_other_uses(tb_demand: Table, tb_total: Table) -> Table:
+    # Add "Other uses" from sheet 1 to the demand table.
+    error = f"Technology category '{OTHER_LABEL}' not found in sheet 1."
```

Comment:
why define these before throughout code instead of just doing this? -     assert OTHER_LABEL in set(tb_total[tb_total["technology"].str.lower().str.contains("other")]["technology"]), f"Technology category '{OTHER_LABEL}' not found in sheet 1."

Reply:
I usually do that for clarity, because the assertion line become very long and hard to interpret, but it's not common practice. It helps read the intention of the assertion before needing to interpret what the assertion does. In a way, it replaces a comment that would say almost exactly the same that the error message.